### ISSUE TITLE
[FEATURE] OPE Benchmark harness (skdr_eval.benchmarks) with public-dataset adapters [M3]

### ISSUE THREAD

**Issue by dgenio** [dgenio/skdr-eval#94]
# ✨ Feature Request

## 📋 Summary

Build a first-class **benchmark harness** in `skdr_eval.benchmarks` — a `BenchmarkSpec(dataset, logging_policy, target_policy, estimators)` runner that produces a comparable estimator table, ships at least two synthetic and one public-dataset benchmark, and maintains a `BENCHMARKS.md` table updated automatically on each release. This is the **concrete executable design + initial implementation** for the umbrella open issue **#31**.

### Use case
- An OPE researcher comparing skdr-eval against OBP / SCOPE-RL needs a one-line answer to "what's your MSE on Open Bandit Dataset?".
- A maintainer needs a regression guard against estimator drift across releases.
- A new contributor adding an estimator (#7 MRDR, #6 MIPS) needs a standard way to demonstrate it's no worse than DR.

### Current limitations
- No public benchmark numbers; credibility gap vs OBP/SCOPE-RL.
- No CI guard against silent estimator regressions across releases.
- #31 names the goal but has no runnable design.

### Module structure
```
src/skdr_eval/benchmarks/
    __init__.py
    spec.py            # BenchmarkSpec dataclass + Runner
    datasets/
        __init__.py
        synthetic.py   # heavy_tail, autoscaling, large_action_space
        open_bandit.py # Open Bandit Dataset adapter (lazy download, cached)
        criteo.py      # Criteo BLBF adapter (placeholder, opt-in)
    policies/
        uniform.py
        sklearn.py
        topk.py
    report.py          # renders BENCHMARKS.md
```

### Public API
```python
from skdr_eval.benchmarks import BenchmarkSpec, run, Synthetic, OpenBandit

spec = BenchmarkSpec(
    dataset=Synthetic.heavy_tail(n=10000, seed=42),
    logging_policy="uniform",
    target_policies={"hgb_pi": "sklearn:HistGradientBoostingRegressor"},
    estimators=["DR", "SNDR", "MRDR", "SWITCH", "DRos", "MIPS"],
    n_reps=200,
)
results = run(spec)
# Returns: DataFrame[estimator, mean_V_hat, MSE_vs_truth, bias, variance, mean_ESS]
```

### CLI integration
```bash
skdr-eval benchmark --spec specs/heavy_tail.yaml --out BENCHMARKS.md
make benchmark   # runs the full suite, ≤30 min on a laptop
```

### BENCHMARKS.md auto-update
- `make benchmark` regenerates the table.
- CI nightly job runs the benchmark on the smallest spec; failure on >2× MSE regression vs main.

### Files touched
- `src/skdr_eval/benchmarks/` (new package)
- `Makefile` — `benchmark` target
- `.github/workflows/benchmark.yml` (new) — nightly + on-tag job
- `BENCHMARKS.md` (new) — generated artifact, committed
- `pyproject.toml` — new `[benchmarks]` extra (for the Open Bandit / Criteo download dependencies)
- `examples/09_benchmark_my_estimator.ipynb` (new)

## ✅ Acceptance Criteria

- [ ] `skdr_eval.benchmarks.run(spec)` returns a comparable estimator-by-metric DataFrame.
- [ ] Three benchmarks shipped: 2 synthetic (heavy-tail, large-action-space) + 1 public (Open Bandit Dataset, lazy-downloaded under `[benchmarks]` extra).
- [ ] `BENCHMARKS.md` committed to repo and regenerated by `make benchmark`.
- [ ] Nightly CI job runs the smallest spec and fails on MSE regression > 2× rolling-3-day median.
- [ ] CLI `skdr-eval benchmark --spec <yaml> --out <md>` works (gated on #10 CLI).
- [ ] At least one benchmark is reproducible from the README in under 5 minutes.
- [ ] No new top-level deps (gated behind `[benchmarks]` extra).

## 🧪 Testing Strategy

- Unit: `BenchmarkSpec` validation (fail-loud on missing fields).
- Smoke: each shipped synthetic benchmark runs to completion in ≤2 min.
- Reproducibility: same seed → identical numbers across runs (within 1e-10).
- Regression: golden snapshot of synthetic-benchmark results checked into tests; bumped intentionally.

## 📚 References

- Saito et al., *Open Bandit Pipeline / Open Bandit Dataset*: https://github.com/st-tech/zr-obp + https://research.zozo.com/data.html
- SCOPE-RL benchmarking design: https://github.com/hakuhodo-technologies/scope-rl
- Criteo Counterfactual Learning (BLBF): https://www.cs.cornell.edu/~adith/Criteo/

## 📎 Related Issues

- **Concrete implementation for #31** — propose to close #31 in favor of this.
- **Consumes** #7 (composable estimators), #6 (MIPS), #70 (public dataset loaders), #9 (card schema).
- **Feeds** the v1.0 "Differentiators" narrative.

## 🚀 Milestone & Effort

- **Milestone:** M3 Differentiators
- **Estimated effort:** Large (3–4 weeks; biggest cost is the Open Bandit adapter + nightly CI plumbing)
- **Breaking changes:** None (new package; gated behind `[benchmarks]` extra).

**Comment by dgenio** [comment]
Growth/adoption refinement:

The benchmark harness should produce a public artifact that can be linked from README, docs, launch posts, and comparison pages. For adoption, the key output is not just a runner API; it is a credible, reproducible evidence page.

Suggested additions to acceptance criteria:

- `BENCHMARKS.md` includes a short human-readable summary before the tables: what was tested, what the numbers mean, and what should not be inferred.
- Include at least one comparison row against a known ecosystem baseline where feasible, e.g. OBP-compatible DR/SNDR behavior on an adapted dataset.
- Publish a small `make benchmark-smoke` path that finishes quickly and is suitable for CI/docs examples.
- Add a README badge/link or section once the benchmark is stable.

The adoption goal is to support a claim like: "standard DR/SNDR estimates, plus time-aware folds, support diagnostics, calibration checks, and stakeholder artifacts" without overselling statistical guarantees.
